### DISCUSSION TITLE
Better contribution with dev containers

### DISCUSSION THREAD

**Discussion by paule96** [microsoft/aspire#3332] [Ideas]
Aspire is super fun, and I have started to contribute to it. The setup for a dev machine is kinda a step that always takes a long time and can hold you back from contribution.
Therefore it would be nice to have a devcontainer configuration in this repository to just spin up a local (or browser-based) devcontainer in like 10 minutes and start working on it. I have the files already created for my contribution.
Would the contribution of a devcontainer configuration be welcomed here in the repository or are there points against it?
(at least for me the debugging of the Dashboard didn't work for example)

**Comment by davidfowl** [comment]
I think it would be welcome. Do we have any prior art @danmoseley @joperezr ?

**Reply by joperezr** [reply]
Hello @paule96, thanks a lot for the suggestion! Yeah, we do have devcontainers setup in multiple repositories, including aspnetcore: 

https://github.com/dotnet/aspnetcore/tree/main/.devcontainer

We haven't really considered it for aspire too, but it would make sense to have it as well. I'd imagine we would have something similar to that. Is that more or less what you had in mind?

Also, I don't know if aspnetcore does it too, but I believe that dotnet/runtime has prebuilts as well and they build every night, which helps contributors get a working environment that can build in little to no time. We can consider that as well for aspire.

**Reply by danmoseley** [reply]
Enabling codespaces is a great idea. Prebuilt seems like a nice to have, as build is pretty fast in this repo.

https://github.com/dotnet/runtime/tree/main/.devcontainer is another example (probably less useful than the aspnetcore one)

Do you want to contribute this, @paule96 ? I'm not sure how much you need commit access, but you could give it a go. Happy to flip settings on this end/and or try your PR if needed.

**Reply by paule96** [reply]
@danmoseley yes I just want to ask befor open a random pull request out of no where. But yes I can contribute this quite easily. As mentioned currently with the downside that you can't debug the dashboard with the dev container right now. (I have no glue why)
But that is maybe a discussion for the pull request.

**Reply by danmoseley** [reply]
@kvenkatrajan is this something you'd expect to work -- your team tested codespaces?

@paule96 a PR would be welcome for the experience we can have today, it can always be improved later.

**Reply by paule96** [reply]
@danmoseley @davidfowl @joperezr here it is the PR:
https://github.com/dotnet/aspire/pull/3520
